core/state_processor.go

```diff
@@ -98,13 +119,12 @@ func applyTransaction(config *chain.Config, engine consensus.EngineReader, gp *G
 		if msg.To() == nil {
 			receipt.ContractAddress = crypto.CreateAddress(evm.TxContext().Origin, tx.GetNonce())
 		}
-		// Set the receipt logs and create a bloom for filtering
+
 		receipt.Logs = ibs.GetLogs(tx.Hash())
-		receipt.Bloom = types.CreateBloom(types.Receipts{receipt})
```

Comment:
Seems like instead of removing this line, if we're looking to leave compatibility, we should have guarded it with a `if !isBobaLegacyBlock` ?

Reply:
It's moved to the later code, because we need to insert our legacy receipt before creating the `Bloom`.

Reply:
My mistake -- I saw the bloom creation below, but thought it was guarded by `isBobaLegacyBlock` (but it's not).